### PR TITLE
logic for scuntToken creation and user updates

### LINKED ISSUE #396: Token
Generate a random 5 digit token for each user
This will be used to login to Scunt using the discord bot - since passwords aren't safe

Frosh will use their email and token to sign in

Can you also create an API call that returns their Scunt team (which should be undefined right now) if the user successfully enters the code that matches their email

If the email doesn't exist, return email not found
If the code is incorrect, return invalid code

This token code will be on the Frosh's Profile - please tie this to the user object so it can be shown on the frontend

### LINKED ISSUE #415: Endpoint for Discord Scunt
When an email and token matches that of an accounts info, set isLoggedInToDiscord to true for that account
Return their Scunt Team when success
and if it was already set to true - return that error

### PR DESCRIPTION
**Pull request by Emily9023**
user will have a token stored 

user inputs email and token on discord to log into scunt - if token and email match the records, then isLoggedInOnDiscord is set to true (default: false) and returns their scunt team

If the email doesn't exist, return email not found

If the code is incorrect, return invalid code

TODO:

- [x] debugging
- [x] testing
- [x] create API to send back information

### CODE REVIEW

File: server/src/services/UserServices.js [hunk 1]
```diff
@@ -86,6 +86,32 @@ const UserServices = {
...
   },
 
+  async checkScuntToken(existingUser) {
+    if (!existingUser.scuntToken && existingUser.isRegistered == true && existingUser.scunt) {
+      return false;
+    }
+    return true;
+  },
+
+  async getScuntToken(email) {
```
Comment (Cali0707) [inline server/src/services/UserServices.js:96]: It looks like this is adding a scunt token to a user. If that is the case, it should be renamed to reflect the fact that it is adding, not getting, a token for a user

**Review by Cali0707 — CHANGES_REQUESTED** [review]
Overall this is good, just some changes need to be made to keep the code consistent with the rest of the codebase

File: server/src/controllers/ScuntController.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        const updateScuntLogin = { isScuntDiscordLoggedIn: true };
+        await UserServices.updateUserInfo(existingUser.id, updateScuntLogin);
+      } catch (err) {
+        res.status(400).send({ message: 'ERROR_UPDATING_USER' });
+        next(err);
+      }
+
+      const userInfo = {
+        name: existingUser.firstName,
+        teamNumber: 'undefined', // existingUser.teamNumber
```
[the remark this answers is not in the document]
Comment (Emily9023) [reply server/src/controllers/ScuntController.js:32]: i think we havent established teamNumber yet for Users so its just a placeholder

[the remark this answers is not in the document]
Comment (jameskokoska) [reply server/src/controllers/ScuntController.js:31]: To be consistent with javascript either leave blank or set as undefined (not a string)

[the remark this answers is not in the document]
Comment (Cali0707) [reply server/src/controllers/ScuntController.js:31]: @Emily9023 please update `teamNumber` to be `null` or `undefined` here if there is no number yet

**Review by Cali0707 — CHANGES_REQUESTED** [review]

File: server/src/services/UserServices.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ const bcrypt = require('bcrypt');
 const EmailValidator = require('email-validator');
 const jwt = require('jsonwebtoken');
 const mongoose = require('mongoose');
+const getScuntToken = require('../middlewares/createScuntToken');
```
Comment (Freeassassin) [inline server/src/services/UserServices.js:5]:
```suggestion
```

This is causing a crash in the backend.

**Review by Freeassassin — APPROVED** [review]
LGTM, just fix the one line!

File: server/src/controllers/UserController.js [hunk 4]
```diff
@@ -56,8 +58,13 @@ const UserController = {
...
   async getInfo(req, res) {
-    const user = req.user.getResponseObject();
-    res.status(200).send({ user });
+    const user = req.user;
+
+    if (!(await UserServices.checkScuntToken(user))) {
+      await UserServices.addScuntToken(user.email.toLowerCase());
+    }
+
+    res.status(200).send(user.getResponseObject());
```
Comment (Cali0707) [inline server/src/controllers/UserController.js:67]:
```suggestion
    let user = req.user;

    if (!(await UserServices.checkScuntToken(user))) {
      user = await UserServices.addScuntToken(user.email.toLowerCase());
    }

    res.status(200).send(user.getResponseObject());
```

**Review by Cali0707 — CHANGES_REQUESTED** [review]

**Review by Cali0707 — APPROVED** [review]

**Pull request merged**